### PR TITLE
Fixes #104

### LINKED ISSUE #104: Derivation doesn't work for hierarchies starting from `sealed abstract class`
I observed this in relation with pprint which doesn't work (i.e. falls back to `toString`) for `Either[X, Y]` values because `Either` is defined as an `sealed abstract class`.

### PR DESCRIPTION
**Pull request by Voltir**
Abstract classes are treated like traits as far as Derive is concerned.

### CODE REVIEW

File: test/Main.scala [hunk 1]
```diff
@@ -2,6 +2,6 @@
 object Main{
   def main(args: Array[String]): Unit = {
     val x = derive.PythonAst
-    implicitly[upickle.default.Writer[derive.PythonAst.expr]]
+    //implicitly[upickle.default.Writer[derive.PythonAst.expr]]
```
Comment (lihaoyi) [inline test/Main.scala]: What's this though?

**Comment by lihaoyi** [comment]
Looks fine, except for that commented out thing

**Comment by Voltir** [comment]
That appears to be the source of the build instability I was talking about on gitter. Uncommenting that line and then running clean compile in sbt would fail with a compile time error every half dozen attempts or so.

**Comment by lihaoyi** [comment]
This looks good. I'll look into the PythonAst stuff offline.

Thanks!

**Pull request merged**